gcloud.conf

```diff
@@ -43,6 +43,8 @@ ES_DOWNLOAD_URL=$( curl $CURL_ARGS -H "Metadata-Flavor: Google" "http://metadata
 [[ $DEBUG ]] && echo ES_DOWNLOAD_URL=$ES_DOWNLOAD_URL || true
 CUSTOM_ES_JAVA_OPTS=$( curl $CURL_ARGS -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/attributes/custom_es_java_opts" )
 [[ $DEBUG ]] && echo CUSTOM_ES_JAVA_OPTS=$CUSTOM_ES_JAVA_OPTS || true
+DATA_DEVICE=$( curl $CURL_ARGS -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/attributes/es_data_device" )
+[[ $DEBUG ]] && echo DATA_DEVICE=$DATA_DEVICE || true
```

Comment:
Just for my own info.
curl -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/attributes/custom_es_java_opts - returns some gibberish  HTML.
What information should this return?

Reply:
It returns the metadata of the machine that you called it from, based on source IP. The metadata is set in spawner.sh on line 228, and read here. This is how you pass parameters into the individuals slaves.